receiver/flinkmetricsreceiver/testdata/expected_metrics/metrics_golden.json

```diff
@@ -0,0 +1,1144 @@
+                        ]
+                     },
+                     "unit": "By"
+                  },
+                  {
+                     "description": "The total number of live threads.",
+                     "name": "flink.jvm.threads.count",
+                     "sum": {
+                        "aggregationTemporality": "AGGREGATION_TEMPORALITY_CUMULATIVE",
+                        "dataPoints": [
+                           {
+                              "asInt": "6",
+                              "startTimeUnixNano": "1653576152546694000",
+                              "timeUnixNano": "1653576152546733000"
+                           }
+                        ]
+                     },
+                     "unit": "{threads}"
+                  },
+                  {
+                     "description": "The total amount of managed memory.",
+                     "name": "flink.memory.managed.total",
+                     "sum": {
+                        "aggregationTemporality": "AGGREGATION_TEMPORALITY_CUMULATIVE",
+                        "dataPoints": [
+                           {
+                              "asInt": "28",
+                              "startTimeUnixNano": "1653576152546694000",
+                              "timeUnixNano": "1653576152546733000"
+                           }
+                        ]
+                     },
+                     "unit": "By"
+                  },
+                  {
+                     "description": "The amount of managed memory currently used.",
+                     "name": "flink.memory.managed.used",
+                     "sum": {
+                        "aggregationTemporality": "AGGREGATION_TEMPORALITY_CUMULATIVE",
+                        "dataPoints": [
+                           {
+                              "asInt": "29",
+                              "startTimeUnixNano": "1653576152546694000",
+                              "timeUnixNano": "1653576152546733000"
+                           }
+                        ]
+                     },
+                     "unit": "By"
+                  }
+               ],
+               "scope": {
+                  "name": "otelcol/flinkmetricsreceiver",
+                  "version": "latest"
+               }
+            }
+         ]
+      },
+      {
+         "resource": {
+            "attributes": [
+               {
+                  "key": "host.name",
+                  "value": {
+                     "stringValue": "mock-host"
+                  }
+               },
+               {
+                  "key": "flink.taskmanager.id",
+                  "value": {
+                     "stringValue": "mock-taskmanager-id"
+                  }
+               },
+               {
+                  "key": "flink.resource.type",
+                  "value": {
+                     "stringValue": "taskmanager"
+                  }
+               }
+            ]
+         },
```

Comment:
As noted elsewhere, there seems to be a major bug in the code where it is impossible to emit more than one taskmanager resource. The same is true for jobs.

I think it would make sense to include a second of each resource in the expected result, to be sure this doesn't slip past or regress in the future.

Reply:
added a second taskmanager and job. updated